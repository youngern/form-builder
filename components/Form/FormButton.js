import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import { Colors } from 'react-native/Libraries/NewAppScreen';

const FormButton = ({ label, style, labelStyle = {}, ...props }) => (
  <TouchableOpacity style={[styles.button, style]} {...props}>
    <Text style={[styles.buttonText, labelStyle]}>{label}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  button: {
    padding: 10,
    borderWidth: 1,
    flex: 1,
    margin: 10,
    borderRadius: 5,
    borderColor: 'lightgray',
    alignItems: 'center',
    backgroundColor: Colors.white,
  },

  buttonText: {
    fontWeight: 'bold',
    color: Colors.white,
  },
});

export default FormButton;
