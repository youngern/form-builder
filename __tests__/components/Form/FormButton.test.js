import React from 'react';
import FormButton from '~/src/components/Form/FormButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <FormButton
        label="Add Field"
        onPress={() => {}}
        style={{
          backgroundColor: '#fbfefd',
          borderColor: '#679b9b',
          borderWidth: 2,
        }}
        labelStyle={{ fontSize: 20, color: '#679b9b' }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without any styling', () => {
  const tree = renderer
    .create(<FormButton label="Add Field" onPress={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
