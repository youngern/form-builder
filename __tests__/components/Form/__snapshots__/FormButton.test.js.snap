// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fbfefd",
      "borderColor": "#679b9b",
      "borderRadius": 5,
      "borderWidth": 2,
      "flex": 1,
      "margin": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontWeight": "bold",
        },
        Object {
          "color": "#679b9b",
          "fontSize": 20,
        },
      ]
    }
  >
    Add Field
  </Text>
</View>
`;

exports[`renders correctly without any styling 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderColor": "lightgray",
      "borderRadius": 5,
      "borderWidth": 1,
      "flex": 1,
      "margin": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontWeight": "bold",
        },
        Object {},
      ]
    }
  >
    Add Field
  </Text>
</View>
`;
